api/v1alpha1/virtualmachinepublishrequest_types.go

```diff
@@ -0,0 +1,530 @@
+	//
+	// +optional
+	Source VirtualMachinePublishRequestSource `json:"source,omitempty"`
+
+	// Target is the target of the publication request, ex. item
+	// information and a ContentLibrary resource.
+	//
+	// If this value is omitted, the controller uses spec.source.name + "-image"
+	// as the name of the published item. Addtionally, when omitted the
+	// controller attempts to identify the target location by matching a
+	// resource with an API version equal to spec.target.location.apiVersion, a
+	// kind equal to spec.target.location.kind, w/ the label
+	// "imageregistry.vmware.com/default".
+	//
+	// Please note that while optional, if a VirtualMachinePublishRequest sans
+	// target information is applied to a namespace without a default
+	// publication target, then the VirtualMachinePublishRequest resource
+	// will be marked in error.
+	//
+	// +optional
+	Target VirtualMachinePublishRequestTarget `json:"target,omitempty"`
+
+	// TTLSecondsAfterFinished is the time-to-live duration for how long this
+	// resource will be allowed to exist once the publication operation
+	// completes. After the TTL expires, the resource will be automatically
+	// deleted without the user having to take any direct action.
+	//
+	// If this field is unset then the request resource will not be
+	// automatically deleted. If this field is set to zero then the request
+	// resource is eligible for deletion immediately after it finishes.
+	//
+	// +optional
+	TTLSecondsAfterFinished *int64 `json:"ttlSecondsAfterFinished,omitempty"`
+}
+
+// VirtualMachinePublishRequestStatus defines the observed state of a
+//  VirtualMachinePublishRequest.
+type VirtualMachinePublishRequestStatus struct {
+	// CompletionTime represents time when the request was completed. It is not
+	// guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	//
+	// The value of this field should be equal to the value of the
+	// LastTransitionTime for the status condition Type=Complete.
+	//
+	// +optional
+	CompletionTime metav1.Time `json:"completionTime,omitempty"`
+
+	// StartTime represents time when the request was acknowledged by the
+	// controller. It is not guaranteed to be set in happens-before order
+	// across separate operations. It is represented in RFC3339 form and is
+	// in UTC.
+	//
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+
+	// ImageName is the name of the VirtualMachineImage resource that is
+	// eventually realized in the same namespace as the VM and publication
+	// request after the publication operation completes.
+	//
+	// This field will not be set until the VirtualMachineImage resource
+	// is realized.
+	//
+	// +optional
+	ImageName string `json:"imageName,omitempty"`
+
+	// Ready is set to true only when the VM has been published successfully
+	// and the new VirtualMachineImage resource is ready.
```

Comment:
Do we have (or plan to add) any readiness checks on our VirtualMachineImage resource? If not, feel like "exist" or "realized" sounds more accurate than "ready".

Reply:
Ready is a standard status term. It means the VM Image was published and ready for use.